### PR TITLE
:sparkles: Add Risk Module

### PR DESCRIPTION
**Pull request by zekiblue**
- Add new library for bitmap for assets

### CODE REVIEW

File: src/libraries/AssetConfiguration.sol [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        //16 bit (32-47) : Liquidity reward
+        //16 bit (48-63) : Protocol liquidation fee
+        //8  bit (64-71) : Decimals
+        //1  bit (72)    : Asset is active
+        //1  bit (73)    : Asset is frozen
+        //1  bit (74)    : Asset is paused
+        //1  bit (75)    : Asset borrowing is enabled
+        //52 bit (76-127): Unused
+        uint128 data;
+    }
```
Comment (zekiblue) [inline src/libraries/AssetConfiguration.sol:29]: @Thomas-Smets @simluyt @j-vp Can there be more parameters added here?

Comment (j-vp) [reply src/libraries/AssetConfiguration.sol:29]: At least 128 bits left in the same storage slot, so you can fill that up for sure

File: src/libraries/AssetConfiguration.sol [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+library AssetConfiguration {
+
+    struct AssetDetailBitmap {
+        // BITMAP
+        //16 bit (0-15)  : Collateral Ratio
+        //16 bit (16-31) : Liquidity Threshold
+        //16 bit (32-47) : Liquidity reward
+        //16 bit (48-63) : Protocol liquidation fee
+        //8  bit (64-71) : Decimals
+        //1  bit (72)    : Asset is active
```
Comment (j-vp) [inline src/libraries/AssetConfiguration.sol:23]: Define what active/paused/... is

File: src/test/unit/libraries/AssetConfiguration.t.sol [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        AssetConfiguration.setCollateralFactor(initialConfig, uint128(1));
+
+        // When: The collateral factor is fetched
+        uint256 collateralFactor2 = AssetConfiguration.getCollateralFactor(initialConfig);
+
+        // Then: New collateral factor should be 1 not 0
+        assertEq(collateralFactor2, 1);
+        assertTrue(collateralFactor2 != 0);
+
+    }
```
Comment (zekiblue) [inline src/test/unit/libraries/AssetConfiguration.t.sol:44]: @alikonuk1 an example for given, when, then testing comments

Comment (Thomas-Smets) [reply src/test/unit/libraries/AssetConfiguration.t.sol:44]: I like it

File: src/libraries/AssetConfiguration.sol (hunk 2 above)
Comment (zekiblue) [reply src/libraries/AssetConfiguration.sol:23]: Ill add the details to get set functions natspecs

**Comment by zekiblue** [comment]
> I think there are still quite some functions missing, in my idea, the risk module should be able to calculate the collateral threshold and liquidation threshold of a vault given the value of the vault per credit rating.
> 
> Some other remarks:
> 
> * Why do we need the decimals of the asset in the risk module?
> * Do we sneed to store a collateral threshold and liquidation threshold per asset or per credit rating?
> * I don't think it is a good idea that every asset we allow now has to be set two times: once in the subregistry and once in the risk module. Ideally the asset is. We have to align what the purpose of the subregistry is and what the purpose of the risk module

Good point, ill add collateral value and liquidation related calculations to the risk module. 

* I added decimals because there were a lot of empty bits in the bitmap, and i thought might be useful later. And in collateral value calculations can be useful now.
* I think it's better to keep the collateral factor and liquidation threshold per asset, in case the asset deviates from the credit rating to the riskier part, we can do emergency tailored actions
* Yes I agree there has to be one definition per asset configuration, ideally we can merge subregistry configuration into struct (as a datatype) and use in libraries and/or in contracts. And subregistries can be more asset general configuration with risk module and pricing module used in it.

**Pull request closed without merging**